tests/test_entities.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from collections import OrderedDict
+from misc import Vector
+
+
+@pytest.fixture
+def player():
+    from entities import Actor
+    from behavior import NullBehavior
+    player = Actor('Player', Vector(0, 0), NullBehavior(), '@', (255, 255, 255))
+    player.backpack.contents = OrderedDict()
+    player.backpack.cur_weight = 0.0
+    return player
+
+
+@pytest.fixture
+def orc():
+    from entities import Actor
+    from behavior import BasicMonster
+    return Actor('Orc', Vector(5, 5), BasicMonster(), 'o', (0, 255, 0))
+
+
+@pytest.fixture
+def gold():
+    from entities import Item
+    return Item('gold', Vector(1, 0), 'g', (0, 125, 125), weight=0.0)
+
+
+@pytest.fixture
+def potion():
+    from entities import Item
+
+    def heal(target):
+        target.hp += 25
+
+    return Item('potion', Vector(0, 1), 'p', (255, 0, 0), effect=heal)
+
+
+@pytest.fixture
+def game_map(player, orc):
+    from display_manager import DisplayManager
+    display_manager = DisplayManager
+    display_manager.gen_map(player)
+    display_manager.cur_map.entities += [player, orc]
+    return display_manager.cur_map
+
+
+class TestEntity(object):
+
+    def test_distance_to(self, player, gold, potion):
+        assert player.distance_to(gold.pos) == player.distance_to(potion.pos)
+
+    def test_stat_recompute(self, player):
+        player.strength += 1
+        assert player.physical_dmg == 3.96
+
+    def test_actor_death(self, player):
+        player.hp -= 200
+        assert player.hp == 0
+        assert player.dead
+
+    def test_actor_level_up_standard(self, player):
+        player.exp += 83
+        assert player.level == 2
+        assert player.exp == 0
+
+    def test_actor_level_up_overshoot(self, player):
+        player.exp += 166
+        assert player.level == 3
+        assert player.exp == 0
+
+    def test_actor_level_up_extra(self, player):
+        player.exp += 100
+        assert player.level == 2
+        assert player.exp == 17
+
+
+class TestBackpackItem(object):
+
+    def test_add_to_backpack(self, player, gold):
+        player.backpack.add(gold, 100)
```

Comment:
If we're gonna treat gold like an item in the backpack, maybe we should remove the gold attribute in Actor.

Reply:
No I just used it as an example item, might as well call it `foo`